lib/dry/system/settings.rb

```diff
@@ -29,9 +29,18 @@ def call
           end
         end
 
-        def key(name, type)
+        def setting(name, type)
+          if ::Kernel::__callee__ == :key
+            ::Dry::Core::Deprecations.announce(
+              "the name of this method will be renamed to :settings",
+              "Please refactor your call at #{::Kernel::caller[0]}",
+              tag: "dry-system",
+              uplevel: 2
+            )
+          end
           schema[name] = type
         end
+        alias :key :setting
```

Comment:
There's actually a `deprecate` macro in `Deprecations` extension that you could use here https://github.com/dry-rb/dry-core/blob/master/lib/dry/core/deprecations.rb#L152-L157

Reply:
Thanks for your hint!
Since the DSL class is inheriting from BasicObject and is not including any Dry module I don't have access to the deprecate call. Can you point me to a case where there is a similar case so I can take a look on how to do this?

Reply:
Check out https://github.com/dry-rb/dry-system/blob/176ead290bb5018bd8e6b5a69226eda51392b5ed/lib/dry/system/container.rb#L98